syft/generic/id_provider.py

```diff
@@ -13,7 +17,9 @@ class IdProvider:
 
     def __init__(self, given_ids=list()):
         self.given_ids = given_ids
-        self.generated = []
+        self.generated = (
+            []
+        )  # TODO: use sorted container to quickly check whether a value is contained
```

Comment:
Can we use a set before merging this to dev?

Reply:
+1

Reply:
I tried, but it is not as easy as it seems. There is a mechanism to retrieve the list of generated ids via a ResponseSignatureError. And the order of the elements in the list is used to assign ids to return values. So simply changing self.generated to a set() results in errors in the unittests with remote method calls that have more than one return value.

Reply:
The generated ids are now stored in a set.